import_obfuscation.go

```diff
@@ -0,0 +1,473 @@
+						privateImports = append(privateImports, importPathCombos(am.Imports[i].Pkg)...)
+					}
+					am.Imports[i].Pkg = hashImport(am.Imports[i].Pkg, garbledImports)
+				}
+			}
+			for i := range am.Packages {
+				if isPrivate(am.Packages[i]) {
+					privateImports = append(privateImports, am.Packages[i])
+					if strings.ContainsRune(am.Packages[i], '/') {
+						privateImports = append(privateImports, importPathCombos(am.Packages[i])...)
+					}
+					am.Packages[i] = hashImport(am.Packages[i], garbledImports)
+				}
+			}
+
+			// move imports that contain another import as a substring to the front,
+			// so that the shorter import will not match first and leak part of an
+			// import path
+			sort.Slice(privateImports, func(i, j int) bool {
+				iSlashes := strings.Count(privateImports[i], "/")
+				jSlashes := strings.Count(privateImports[j], "/")
+				if iSlashes == 0 && jSlashes == 0 {
+					return privateImports[i] > privateImports[j]
+				}
+				return iSlashes > jSlashes
+			})
+			privateImports = dedupImportPaths(privateImports)
+
+			// no private import paths, nothing to garble
+			// log.Printf("== Private imports: %v ==\n", privateImports)
+			if len(privateImports) == 0 {
+				continue
+			}
+
+			// garble all private import paths in all symbol names
+			lists := [][]*goobj2.Sym{am.SymDefs, am.NonPkgSymDefs, am.NonPkgSymRefs}
+			for _, list := range lists {
+				for _, s := range list {
+					// garble read only static data, but not strings. If import paths are in strings,
+					// that means garbling strings might effect the behavior of the compiled binary
+					if int(s.Kind) == 2 && s.Data != nil && !strings.HasPrefix(s.Name, "go.string.") {
+						var dataTyp dataType
+						if strings.HasPrefix(s.Name, "type..importpath.") {
+							dataTyp = importPath
+						} else if strings.HasPrefix(s.Name, "type..namedata.") {
+							dataTyp = namedata
+						}
+						s.Data = garbleSymData(s.Data, privateImports, garbledImports, dataTyp, &buf)
+
+						if s.Size != 0 {
+							s.Size = uint32(len(s.Data))
+						}
+					}
+					s.Name = garbleSymbolName(s.Name, privateImports, garbledImports, &sb)
+
+					for i := range s.Reloc {
+						s.Reloc[i].Name = garbleSymbolName(s.Reloc[i].Name, privateImports, garbledImports, &sb)
+					}
+					if s.Type != nil {
+						s.Type.Name = garbleSymbolName(s.Type.Name, privateImports, garbledImports, &sb)
+					}
+					if s.Func != nil {
+						for i := range s.Func.FuncData {
+							s.Func.FuncData[i].Sym.Name = garbleSymbolName(s.Func.FuncData[i].Sym.Name, privateImports, garbledImports, &sb)
+						}
+						for _, inl := range s.Func.InlTree {
+							inl.Func.Name = garbleSymbolName(inl.Func.Name, privateImports, garbledImports, &sb)
+						}
+					}
+				}
+			}
+			for i := range am.SymRefs {
+				am.SymRefs[i].Name = garbleSymbolName(am.SymRefs[i].Name, privateImports, garbledImports, &sb)
+			}
+
+			if err = p.pkg.Write(p.path); err != nil {
```

Comment:
always do `if err := ...; err != nil {`, so that you don't depend on the parent scope having an err variable, and you don't modify that either. there are some other cases in your PR too.